Python/bytecodes.c

```diff
@@ -2450,7 +2450,7 @@ dummy_func(
             }
             DEOPT_IF(!PyFunction_Check(callable), CALL);
             PyFunctionObject *func = (PyFunctionObject *)callable;
-            DEOPT_IF(func->func_version != func_version, CALL);
+            DEOPT_IF(!_PyFunction_IsSimple(func), CALL);
             PyCodeObject *code = (PyCodeObject *)func->func_code;
             DEOPT_IF(code->co_argcount != argcount, CALL);
```

Comment:
If we keep the version number check, we can convert this to a check on `is_meth`, which might be quicker as it saves reaching into the function object.